controllers/storagecluster/storageclasses.go

```diff
@@ -355,3 +414,37 @@ func (r *StorageClusterReconciler) newStorageClassConfigurations(initData *ocsv1
 
 	return ret, nil
 }
+
+func getTopologyConstrainedPools(initData *ocsv1.StorageCluster) string {
+	failureDomain := initData.Status.FailureDomain
+	failureDomainValues := initData.Status.FailureDomainValues
+
+	type topologySegment struct {
+		DomainLabel string `json:"domainLabel"`
+		DomainValue string `json:"value"`
+	}
+
+	// TopologyConstrainedPool stores the pool name and a list of its associated topology domain values.
+	type constrainedPool struct {
```

Comment:
Can we reference these types in the csi repo instead of redefining them here?

Reply:
What types are you referring to? Is `constrainedPool` a struct used in the Ceph-CSI repo?

Reply:
They are defined [here](https://github.com/ceph/ceph-csi/blob/1952a9b4b3ff4b00859d5b663d2f3d4a7d07542b/internal/util/topology.go#L129-L139), but since one of them is private (lower-case), it might not be worth asking for a change in the csi repo, and just keep them redefined here.